lib/puppet/provider/group/directoryservice.rb

```diff
@@ -9,4 +9,14 @@
   confine :operatingsystem => :darwin
   defaultfor :operatingsystem => :darwin
   has_feature :manages_members
+
+  def members_insync?(current, should)
+    return false unless current
+    if current.respond_to?(:sort) and should.respond_to?(:sort)
```

Comment:
So you'd like me to remove the `respond_to?` check on `should` entirely?

Let me try to re-arrange the logic.